### PR TITLE
[BER, FTWD, 012020] Mohamed Ciwan

### PR DESCRIPTION
**Pull request by ciwan-aram**

### CODE REVIEW

File: starter-code/index.js [hunk 1]
```diff
@@ -1,17 +1,55 @@
...
+    }
+    return item;
+  }
 
-  max() {}
+  max() {
+    if (this.items == 0) {
+      throw new Error('EmptySortedList');
+    }
+    return this.items[this.length - 1];
```
Comment (itstheandre) [inline starter-code/index.js]: Even though this works. Ideally you should have used Math.max
```js
return Math.max(...this.items);
```

File: starter-code/index.js [hunk 2]
```diff
@@ -1,17 +1,55 @@
...
+    }
+    return this.items[0];
+  }
 
-  sum() {}
+  sum() {
+    let total = 0;
+    for (let i = 0; i < this.length; i++) {
+      total += this.items[i];
+    }
```
Comment (itstheandre) [inline starter-code/index.js]: Even though this is correct, the reduce would have been a cleaner approach :)

File: starter-code/index.js [hunk 3]
```diff
@@ -1,17 +1,55 @@
...
-  avg() {}
+  avg() {
+    let total = 0;
+    if (this.items == 0) {
+      throw new Error('EmptySortedList');
+    }
+    for (let i = 0; i < this.length; i++) {
+      total += this.items[i];
+    }
+    return total / this.length;
```
Comment (itstheandre) [inline starter-code/index.js]: You could have DRY(Don't repeat yourself)'ed this.

and just:

```js
return this.sum()/ this.items.length
```

Comment (ciwan-aram) [reply starter-code/index.js]: I tried this but somehow couldn't make it work! Something was missing and I didn't have the time to search. 
Thank you for all your comments :)

**Pull request closed without merging**